src/Database.hs

```diff
@@ -13,29 +15,38 @@ import Database.PostgreSQL.Simple.SqlQQ
 import Database.PostgreSQL.Simple.Types
 import Types
 
-connectApp :: DatabaseConfig -> IO Connection
-connectApp (DBConfig host db user password dir) = connectPostgreSQL $ BS8.pack $ unpack url
+connectApp :: MonadIO m => DatabaseConfig -> m Connection
+connectApp (DBConfig host db user password dir) =
+  liftIO $ connectPostgreSQL $ BS8.pack $ unpack url
```

Comment:
You might find it interesting to use a package (or even an alternative prelude) that makes it easier to do these string conversions.

`classy-prelude` (well really, `mono-traversable`) provides a function `encodeUtf8` and `decodeUtf8` that make it slightly easier to go from Text to ByteString.